python/ray/util/client/server/dataservicer.py

```diff
@@ -33,23 +48,47 @@ def Datapath(self, request_iterator, context):
             logger.error("Client connecting with no client_id")
             return
         logger.debug(f"New data connection from client {client_id}: ")
+        if not self._init(client_id, context):
+            return
```

Comment:
why moving this out is OK?
why we no longer need to give the RESOURCE_EXHAAUSTED status code?

Reply:
WDYM? if `_init()` returns False, it means initialization failed, so we just return from the function, which makes the server close the connection.